contracts/ERC5725.sol

```diff
@@ -27,16 +30,39 @@ abstract contract ERC5725 is IERC5725, ERC721 {
      * @dev See {IERC5725}.
      */
     function claim(uint256 tokenId) external override(IERC5725) validToken(tokenId) {
-        require(ownerOf(tokenId) == msg.sender, "Not owner of NFT");
+        require(
+            ownerOf(tokenId) == msg.sender ||
+                (ownerOf(tokenId) != address(0) && allowance(ownerOf(tokenId), msg.sender) > 0),
+            "ERC5725: not owner of NFT or no permission to spend"
+        );
+
         uint256 amountClaimed = claimablePayout(tokenId);
         require(amountClaimed > 0, "ERC5725: No pending payout");
 
+        if (ownerOf(tokenId) != msg.sender) {
+            _spendAllowance(ownerOf(tokenId), msg.sender, amountClaimed);
+        }
+
         emit PayoutClaimed(tokenId, msg.sender, amountClaimed);
 
         _payoutClaimed[tokenId] += amountClaimed;
         IERC20(payoutToken(tokenId)).safeTransfer(msg.sender, amountClaimed);
     }
 
+    /**
+     * @dev See {IERC5725}.
+     */
+    function increaseClaimAllowance(address spender, uint256 addedValue) external override(IERC5725) {
```

Comment:
The refactor I'm suggesting assumes the approved spender can access all of the funds. 

```
function setClaimApprovalForAll(address spender, bool approved) external; 
function setClaimApproval(address spender, bool approved, uint256 tokenId) external; 
```